com.unity.multiplayer.mlapi/Tests/Runtime/Transport/SIPTransport.cs

```diff
@@ -220,38 +220,52 @@ public override SocketTasks StartServer()
 
     public override void Send(ulong clientId, ArraySegment<byte> data, NetworkChannel channel)
     {
-        // Create copy since MLAPI wants the byte array back straight after the method call.
-        // Hard on GC.
-        byte[] copy = new byte[data.Count];
-        Buffer.BlockCopy(data.Array, data.Offset, copy, 0, data.Count);
-
-        m_Clients[clientId].IncomingBuffer.Enqueue(new Event
+        if (m_LocalConnection != null)
         {
-            Type = NetworkEvent.Data,
-            ConnectionId = m_LocalConnection.ConnectionId,
-            Data = new ArraySegment<byte>(copy),
-            Channel = channel
-        });
+            // Create copy since MLAPI wants the byte array back straight after the method call.
+            // Hard on GC.
+            byte[] copy = new byte[data.Count];
+            Buffer.BlockCopy(data.Array, data.Offset, copy, 0, data.Count);
+
+            m_Clients[clientId].IncomingBuffer.Enqueue(new Event
+            {
+                Type = NetworkEvent.Data,
+                ConnectionId = m_LocalConnection.ConnectionId,
+                Data = new ArraySegment<byte>(copy),
+                Channel = channel
+            });
+        }
     }
 
     public override NetworkEvent PollEvent(out ulong clientId, out NetworkChannel channel, out ArraySegment<byte> payload, out float receiveTime)
     {
-        if (m_LocalConnection.IncomingBuffer.Count == 0)
+        if (m_LocalConnection != null)
```

Comment:
If a player is disconnected m_LocalConnection becomes null, but can still be called since the NetworkManager is not notified of the client disconnecting.  The resolution for this was to return back NetworkEvent.Nothing for the still running NetworkManager that is attempting to poll the SIPTransport for the client in question.

Reply:
similar to above, I'd warn dev here and even expect them to look at their call-stack & code-path when this happens.
please put something into the else branch.

Reply:
@TwoTenPvP Please review the above changes.
SIPTransport.DisconnectLocalClient sets m_LocalConnection to null but never notifies the relative NetworkManager that the client has disconnected which can cause other systems (i.e. SnapShotSystem) to cause a barrage of exceptions when trying to send snapshots from a disconnected client.

Reply:
Same as above